account_tax_analysis/models/account_move_line.py

```diff
@@ -0,0 +1,32 @@
+# -*- coding: utf-8 -*-
+# Copyright 2018 ACSONE SA/NV
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models
+
+
+class AccountMoveLine(models.Model):
+
+    _inherit = 'account.move.line'
+
+    analysis_tax = fields.Char(compute="_compute_analysis_tax", store=True)
+
+    @api.multi
+    @api.depends("tax_line_id",
+                 "tax_ids",
+                 "company_id.partner_id.lang")
+    def _compute_analysis_tax(self):
+        companies = self.mapped("company_id")
+        for company in companies:
+            lines = self.filtered(lambda s, c=company: s.company_id == c)
+            lang_lines = lines
+            if company.partner_id.lang:
+                lang_lines = lines.with_context(lang=company.partner_id.lang)
+            for lang_line in lang_lines:
+
+                # computed value must be assign without a with_context()
+                line = lines.filtered(lambda s, o=lang_line: s == o)
```

Comment:
To avoid these line you can do:
`lines = lines.with_prefetch(self._prefetch)` just after line 21

Reply:
this filtered inside the loop is a performance killer

you could do `for line, lang_line in zip(lines, lang_lines)` (assuming with_context preservers the ordering of the set)

Reply:
`with_context` will preserve the order: `_ids` will reference the same set:

[with_context](https://github.com/odoo/odoo/blob/11.0/odoo/models.py#L4409) will call [with_env](https://github.com/odoo/odoo/blob/11.0/odoo/models.py#L4364), which will call [_browse](https://github.com/odoo/odoo/blob/11.0/odoo/models.py#L4308) which only do a `records._ids = ids` and doesn't recreate a new set.

I change it to the `zip`